src/Domain/DoctrineDonationRepository.php

```diff
@@ -52,7 +51,7 @@ public function allocateMatchFunds(Donation $donation): string
         // is most often empty (for new donations) so this will frequently be 0.00.
```

Comment:
as the allocateMatchFunds function is getting longer maybe this would be a good time to move it out of the repository into a new class just for this function and the ones it calls (`allocateFundsAndPrepareDBChanges` and  `releaseNewlyAllocatedFunds`)